object-cache.php

```diff
@@ -273,31 +278,132 @@ function delete( $id, $group = 'default' ) {
 		return $result;
 	}
 
+	// Gets number from all default servers, replicating if needed
+	function get_max_flush_number( $group ) {
+		$key = $this->key( $this->flush_key, $group );
+
+		$values = array();
+		$size = 19;
+		foreach ( $this->default_mcs as $i => $mc ) {
+			$flags = false;
+			$this->timer_start();
+			$values[ $i ] = $mc->get( $key, $flags );
+			$elapsed = $this->timer_stop();
+
+			if ( empty( $values[ $i ] ) ) {
+				$this->group_ops_stats( 'get_flush_number', $key, $group, null, $elapsed, 'not_in_memcache' );
+			} else {
+				$this->group_ops_stats( 'get_flush_number', $key, $group, $size, $elapsed, 'memcache' );
+			}
+		}
+
+		$max = max( $values );
+
+		if ( ! $max > 0 ) {
+			return false;
+		}
+
+		// Replicate to servers not having the max.
+		$expire = 0;
+		$size = 19;
+		foreach ( $this->default_mcs as $i => $mc ) {
+			if ( $values[ $i ] < $max ) {
+				$this->timer_start();
+				$mc->set( $key, $max, false, $expire );
+				$elapsed = $this->timer_stop();
+				$this->group_ops_stats( 'set_flush_number', $key, $group, $size, $elapsed, 'replication_repair' );
+			}
+		}
+
+		return $max;
+	}
+
+	function set_flush_number( $value, $group ) {
+		$key = $this->key( $this->flush_key, $group );
+		$expire = 0;
+		$size = 19;
+		foreach ( $this->default_mcs as $i => $mc ) {
+			$this->timer_start();
+			$mc->set( $key, $value, false, $expire );
+			$elapsed = $this->timer_stop();
+			$this->group_ops_stats( 'set_flush_number', $key, $group, $size, $elapsed, 'replication' );
+		}
+	}
+
+	function get_flush_number( $group ) {
+		$flush_number = $this->get_max_flush_number( $group );
+		// What if there is no v4 flush number?
+		if ( empty( $flush_number ) ) {
+			// Look for the v3 flush number key.
+			$flush_number = intval( $this->get( $this->old_flush_key, $group ) );
+			if ( !empty( $flush_number ) ) {
+				// Found v3 flush number. Upgrade to v4 with replication.
+				$this->set_flush_number( $flush_number, $group );
+				// Delete v3 key so we can't later restore it and find stale keys.
```

Comment:
I guess it would be a good idea to actually delete the key and test this upgrade case?